rca/shortcourses/tests/test_access_planit.py

```diff
@@ -0,0 +1,261 @@
+
+    def test_xml_fetch_no_venue(self):
+        """ Prove that you must pass blank values as parameters"""
+        query = QueryDict(mutable=True)
+        query.update({"CompanyID": "ROYALC9RCH", "courseIDs": 1})
+        query = query.urlencode()
+        response = requests.get(settings.ACCESS_PLANIT_XML_BASE_URL + query, timeout=5)
+        self.assertEqual(response.text, """Missing parameter: venueIDs.\r\n""")
+        self.assertEqual(response.status_code, 500)
+
+    """ Patch the request module totally to force the timeout so we can test the
+        result of the try/expect.
+
+        Test here that data returned from a timeout is None
+    """
+
+    @mock.patch("rca.shortcourses.access_planit.requests.get")
+    def test_data_if_timeout(self, mock_get):
+        """ If a timeout is caught the xml_data should be None"""
+        logging.disable(logging.CRITICAL)
+        mock_get.side_effect = Timeout
+        data = AccessPlanitXML(course_id=1)
+        xml_data = data.get_data()
+        self.assertEqual(xml_data, None)
+
+    @mock.patch(
+        "rca.shortcourses.access_planit.AccessPlanitXML.fetch_data_from_xml",
+        side_effect=mocked_fetch_data_from_xml,
+    )
+    @mock.patch("rca.shortcourses.access_planit.requests.get")
+    def test_cached_data_if_timeout(self, mock_get, mocked_fetch_data_from_xml):
+        """ Test getting stale cache data with a Timeout failure to get data"""
+        # logging.disable(logging.CRITICAL)
+        ShortCoursePage.objects.create(
+            title=f"Short course 1", path="1", depth="001", access_planit_course_id=1
+        )
+        # Generate the mocked xml data
+        AccessPlanitXML(course_id=1).get_data()
+        cache_key = f"short_course_1"
+        self.assertEqual(cache.get(cache_key), self.expected_data)
+        # Add a timeout for the new data request
+        mock_get.side_effect = Timeout
+        AccessPlanitXML(course_id=1).get_data()
+
+        # Now check the stale cache data is there
+        self.assertEqual(cache.get(cache_key), self.expected_data)
+
+    def test_required_course_id(self):
+        """ The course id is a required field, however if this is changed we want some tests to fail,
+        as it will have unwanted effects, the validation on the model should be able to cast the
+        course value to an integer"""
+        with self.assertRaises(TypeError):
+            ShortCoursePage.objects.create(
+                title=f"Short course should not save", path=100, depth="001"
+            )
+        with self.assertRaises(TypeError):
+            ShortCoursePage.objects.create(
+                title=f"Short course should not save",
+                path=100,
+                depth="001",
+                access_planit_course_id="course id cannot be a string",
+            )
+
+    def test_management_command_fetch_data(self):
+        """ Test that xml with no dates doesn't go in the cache """
```

Comment:
How is the test condition "xml with no dates" being set up in this test?
Is it doing the same as below, testing a non valid course ID?

Reply:
I think it _is_ doing the same as below, it's just checking the data return for a course_id passed that won't return anything. The 'XML with no dates' is basically just no data, but I think I'm going to remove this because it similar the below test.